src/controller/tests/data_model/TestRead.cpp

```diff
@@ -177,14 +179,37 @@ class TestReadInteraction
     static void TestReadHandlerResourceExhaustion_MultipleReads(nlTestSuite * apSuite, void * apContext);
 
 private:
+    CHIP_ERROR OnSubscriptionRequested(app::ReadHandler & aReadHandler, Transport::SecureSession & aSecureSession)
```

Comment:
This does not test actually setting the intervals.  That should be tested, because I think it does not work right now.....

Reply:
Agreed. That would require plumbing the intervals through on the `ReadClient::Callback::OnSubscriptionEstablished`, which I actually forgot to update as part of this change, since clients need to be able to get the final negotiated intervals.